tests/bluetooth/audio/mocks/include/gatt_expects.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MOCKS_GATT_EXPECTS_H_
+#define MOCKS_GATT_EXPECTS_H_
+
+#include <zephyr/bluetooth/gatt.h>
+
+#include "gatt.h"
+#include "expects_util.h"
+
+#define expect_bt_gatt_attr_read_called_once(_conn, _attr, _buf, _buf_len, _offset, _value,        \
+					     _value_len)                                           \
+do {                                                                                               \
```

Comment:
Did you consider implementing these as functions instead of these large macros?

Reply:
I could not use the `IF_NOT_EMPTY` macro then